examples/speechbrain_tutorial/download_data.py

```diff
@@ -0,0 +1,36 @@
+import logging
+import sys
+
+import speechbrain as sb
+from hyperpyyaml import load_hyperpyyaml
+from mini_librispeech_prepare import prepare_mini_librispeech
+
+logger = logging.getLogger(__name__)
+
+if __name__ == "__main__":
+    hparams_file, run_opts, overrides = sb.parse_arguments(sys.argv[1:])
+
+    # Initialize ddp (useful only for multi-GPU DDP training)
+    sb.utils.distributed.ddp_init_group(run_opts)
+
+    # Load hyperparameters file with command-line overrides
+    with open(hparams_file) as fin:
+        hparams = load_hyperpyyaml(fin, overrides)
+
+    # Create experiment directory
+    sb.create_experiment_directory(
+        experiment_directory=hparams["output_folder"],
+        hyperparams_to_save=hparams_file,
+        overrides=overrides,
+    )
+
+    # Data preparation, to be run on only one process.
+    sb.utils.distributed.run_on_main(
+        prepare_mini_librispeech,
+        kwargs={
+            "data_folder": hparams["data_folder"],
+            "save_json_train": hparams["train_annotation"],
+            "save_json_valid": hparams["valid_annotation"],
+            "save_json_test": hparams["test_annotation"],
+        },
+    )
```

Comment:
Seems like this is all reimplemented in `main.py`. Maybe it should be turned into a function that can be reused. Or add an argument in main.py to avoid the later training steps.

Reply:
True ! But I have to use `hparams`, which is defined here. I will simply make a `download` function and make it return the hparams. Then I can call the function both times instead of duplicating code !

Reply:
Done in 3c801219f00ff9ee3776b8ff4e3bcc20964ad513. Thanks!